tests/ts/examples.ts

```diff
@@ -1033,3 +1033,32 @@ const plugin2 = ({} as any) as objection.Plugin;
   takesPersonClass(MyPerson);
   takesModelSubclass(new MyPerson());
 };
+
+// Examples with composite key
+class Interview extends objection.Model {
+  interviewer!: Person;
```

Comment:
technically doesn't affect the typescript, but wouldn't these need to be numbers?

Reply:
That's also an interesting thought: `findById(s)` types for the ID should probably be enforced to match the `idColumn` property of the class.

Reply:
> technically doesn't affect the typescript, but wouldn't these need to be numbers?

Right. Should've defined a relationship for this. I was just trying to utilize the existing model class.

> That's also an interesting thought: `findById(s)` types for the ID should probably be enforced to match the `idColumn` property of the class. (unrelated to this PR)

Tried, but didn't manage to.

When you retrieve a `QueryBuilder` from a model (`SomeModel.query()`), only the `Model` type is passed as the type parameter, instead of the `ModelClass` type.

https://github.com/Vincit/objection.js/blob/9669e9617bb3e0db7162724cbd20b917affd5f20/typings/objection/index.d.ts#L448

The case is the same when you construct a `QueryBuilder` with its static method `forClass`:

https://github.com/Vincit/objection.js/blob/9669e9617bb3e0db7162724cbd20b917affd5f20/typings/objection/index.d.ts#L659-L661

It seems that it is impossible to get the actual `ModelClass`, when what we have is just a generic type `QM in <QM extends Model>`.

So the problem is `QueryBuilder` accepts a `Model` instead of a `ModelClass`. However, if we change this requirement, I don't know whether it will break any existing code. Didn't try that far though.

Meanwhile, it is impossible to access the static property of a generic type parameter by lookup types. So dead end here.

Reply:
No need to try to match the `idColumn`. That would be insane :smile: